### PR TITLE
dla-future: Add 0.5.0

### PR DESCRIPTION
**Pull request by rasolca**

### CODE REVIEW

**Comment by rasolca** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/dla-future/package.py [hunk 1]
```diff
@@ -45,6 +46,16 @@ class DlaFuture(CMakePackage, CudaPackage, ROCmPackage):
         description="Build C API compatible with ScaLAPACK",
     )
 
+    variant("mpi-gpu-aware", default=False, when="@0.5.0:", description="Use GPU-aware MPI.")
+    conflicts("+mpi-gpu-aware", when="~cuda ~rocm", msg="GPU-aware MPI requires +cuda or +rocm")
+
+    variant(
+        "mpi-gpu-force-contiguous",
+        default=True,
+        when="@0:5:0: +mpi-gpu-aware",
```
Comment (albestro) [inline var/spack/repos/builtin/packages/dla-future/package.py]:
```suggestion
        when="@0.5.0: +mpi-gpu-aware",
```

File: var/spack/repos/builtin/packages/dla-future/package.py [hunk 2]
```diff
@@ -45,6 +46,16 @@ class DlaFuture(CMakePackage, CudaPackage, ROCmPackage):
...
     )
 
+    variant("mpi-gpu-aware", default=False, when="@0.5.0:", description="Use GPU-aware MPI.")
+    conflicts("+mpi-gpu-aware", when="~cuda ~rocm", msg="GPU-aware MPI requires +cuda or +rocm")
+
+    variant(
+        "mpi-gpu-force-contiguous",
+        default=True,
+        when="@0.5.0: +mpi-gpu-aware",
+        description="Force communication buffers to be contiguous before communicating.",
```
Comment (msimberg) [inline var/spack/repos/builtin/packages/dla-future/package.py]: I was sure I'd added the word "GPU" in the description here, but it seems not (i.e. this matches the DLA-Future definition). However, maybe it'd be good to say:
```suggestion
        description="Force GPU communication buffers to be contiguous before communicating.",
```

**Review by msimberg — DISMISSED** [review]

File: var/spack/repos/builtin/packages/dla-future/package.py (hunk 2 above)
Comment (haampie) [reply var/spack/repos/builtin/packages/dla-future/package.py]: also consider `-` -> `_`, cause some people complain that `+mpi-gpu` looks too much like `+mpi ~gpu` etc

**Review by albestro — APPROVED** [review]
✅ checksum verified

LGTM

**Review by msimberg — APPROVED** [review]

**Pull request merged**